fineract-provider/src/main/java/org/apache/fineract/portfolio/shareaccounts/service/ShareAccountSchedularServiceImpl.java

```diff
@@ -44,7 +45,7 @@ public void postDividend(final Long dividendDetailId, final Long savingsId) {
                 .orElseThrow();
         final SavingsAccount savingsAccount = this.savingsAccountAssembler.assembleFrom(savingsId, false);
         SavingsAccountTransaction savingsAccountTransaction = this.savingsAccountDomainService.handleDividendPayout(savingsAccount,
-                DateUtils.getBusinessLocalDate(), shareAccountDividendDetails.getAmount(), false);
+                DateUtils.getBusinessLocalDate(), shareAccountDividendDetails.getAmount(), false, ExternalId.empty());
```

Comment:
We should check whether enable-auto-generated-external-id is true or false...

if it is true, we need to generate automatically external id.